tests/test_packages/test_connections/test_p2p_libp2p/base.py

```diff
@@ -413,3 +411,24 @@ def sent_is_echoed_envelope(self, sent: Envelope, echoed: Envelope) -> bool:
         attrs_are_identical = (getattr(sent, a) == getattr(echoed, a) for a in attrs)
         addresses_inverted = (getattr(sent, k) == getattr(echoed, v) for k, v in items)
         return all(attrs_are_identical) and all(addresses_inverted)
+
+    @classmethod
+    def _multiplex_it(cls, connection):
+        multiplexer = Multiplexer([connection], protocols=[MockDefaultMessageProtocol])
+        cls.multiplexers.append(multiplexer)
+        multiplexer.connect()
+        return connection
+
+    @classmethod
+    def make_connection(cls, **kwargs) -> P2PLibp2pConnection:
+        connection = cls._multiplex_it(_make_libp2p_connection(**kwargs))
+        cls.log_files.append(connection.node.log_file)
+        return connection
+
+    @classmethod
+    def make_client_connection(cls, **kwargs) -> P2PLibp2pClientConnection:
+        return cls._multiplex_it(_make_libp2p_client_connection(**kwargs))
+
+    @classmethod
+    def make_mailbox_connection(cls, **kwargs) -> P2PLibp2pMailboxConnection:
+        return cls._multiplex_it(_make_libp2p_mailbox_connection(**kwargs))
```

Comment:
ensures that we assign all connections made in tests to the list of multiplexers - which consequently guarantees disconnection and subprocess termination during teardown